nixpkgs/toolchains/foreign_cc.bzl

```diff
@@ -0,0 +1,108 @@
+load("//nixpkgs:nixpkgs.bzl", "nixpkgs_package")
+load("@bazel_tools//tools/cpp:lib_cc_configure.bzl", "get_cpu_value")
+
+_foreign_cc_nix_build = """
+load("@rules_foreign_cc//toolchains/native_tools:native_tools_toolchain.bzl", "native_tool_toolchain")
+
+filegroup(
+    name = "data",
+    srcs = glob(["bin/**"]),
+)
+native_tool_toolchain(
+    name = "cmake_nix_impl",
+    path = "bin/cmake",
+    target = ":data",
+)
+native_tool_toolchain(
+    name = "make_nix_impl",
+    path = "bin/make",
+    target = ":data",
+)
+native_tool_toolchain(
+    name = "ninja_nix_impl",
+    path = "bin/ninja",
+    target = ":data",
+)
+"""
+
+_foreign_cc_nix_toolchain = """
+toolchain(
+    name = "cmake_nix_toolchain",
+    toolchain = "@{toolchain_repo}//:cmake_nix_impl",
+    toolchain_type = "@rules_foreign_cc//toolchains:cmake_toolchain",
+    exec_compatible_with = {exec_constraints},
+    target_compatible_with = {target_constraints},
+)
+toolchain(
+    name = "make_nix_toolchain",
+    toolchain = "@{toolchain_repo}//:make_nix_impl",
+    toolchain_type = "@rules_foreign_cc//toolchains:make_toolchain",
+    exec_compatible_with = {exec_constraints},
+    target_compatible_with = {target_constraints},
+)
+toolchain(
+    name = "ninja_nix_toolchain",
+    toolchain = "@{toolchain_repo}//:ninja_nix_impl",
+    toolchain_type = "@rules_foreign_cc//toolchains:ninja_toolchain",
+    exec_compatible_with = {exec_constraints},
+    target_compatible_with = {target_constraints},
+)
+"""
+
+def _nixpkgs_foreign_cc_toolchain_impl(repository_ctx):
+    exec_constraints = [ "@io_tweag_rules_nixpkgs//nixpkgs/constraints:support_nix", "@platforms//cpu:x86_64" ]
+    target_constraints = [ "@platforms//cpu:x86_64" ]
+    cpu = get_cpu_value(repository_ctx)
```

Comment:
You can use `_ensure_constraints` as shown [here](https://github.com/tweag/rules_nixpkgs/blob/05a445575de51872f528c537cc7ff3ab11114f21/nixpkgs/nixpkgs.bzl#L1028).